### ISSUE TITLE
Compiler doesn't warn about incompatible signatures

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/TypeScript#59922]
### 🔎 Search Terms

is not assignable to type

### 🕗 Version & Regression Information

Seems to be uncaught in all version of TS

### ⏯ Playground Link

https://www.typescriptlang.org/play/?#code/MYGwhgzhAEDKAuAnApmAttA3gKGn6ADgK4BGIAlsNCmACYAUAlAFzQBuA9ubQNy77EylaqgZoOtZKwDkaMAA9pAGmgRyALynQAdkTQlkiFuy69+eQRSo0xE5AH4ZcxSrWbHOvQaOtO3LObQAL7YIdjk2vCGAGZgwMhw8LQRAfgidPTOAEIAnlEQACocAEqirLr6hsZ+ZmHAHNoQ8KpJEawIydrQALw6yADuiTRoTHxNnQB0NvQAjAAMjHxAA

### 💻 Code

```ts
// Your code here
```

### 🙁 Actual behavior

No compile errors are reported

### 🙂 Expected behavior

A compile error about incompatible signatures since a `number` can't be assigned to `'max'`

**Comment by MartinJohns** [comment]
This is working as intended. Methods are bivariant. See the FAQ: [Why Method Bivariance?
](https://github.com/microsoft/TypeScript/wiki/FAQ#why-method-bivariance)
